### ISSUE TITLE
Regarding: Poor documentation is technical debt

### ISSUE THREAD

**Issue by ben-manes** [R-Sandor/CryptoProject#1]
> Here's example of something I dealt with recently with caffeine cache. The original dev didn't understand that fetch is considered a cache write and eviction never occurred. So coming in to fix the bug you couldn't tell whether they misunderstood or if it was intentional. - [tweet](https://twitter.com/codefulDev/status/1592114143192305664)

Hey @R-Sandor, sorry to ping you here but didn't see an easier way to ask. Was the documentation issue a case where I could have done better with Caffeine, unclear internal code, ...? Thanks!

**Comment by R-Sandor** [comment]
Hey Ben! I am really impressed you reached out, I didn't even @ you, when I mentioned on Twitter you do an great job on the project but how I have struggled with some eviction docs in the past.  

I think it was more a failure on my part and the original developer. I think the original developer probably wasn't paying close enough attention. 

One of your posts on SO really helped with my ticket because I noticed that even when the expiration period was hit, it didn't remove it from the cache immediately and you shared how to force the update rather than waiting for another write to the cache.
